c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2836,14 +2838,28 @@ void local_raw_delete(void* ptr);
 
 class NativeCachingAllocator : public CUDAAllocator {
  private:
-  std::mutex mutex;
+  // Shard allocation region to have independent mutexes to reduce contention.
+  static constexpr size_t kNumMutexShard = 67;
```

Comment:
You meant 63? :D

Reply:
63 is not a prime number so I found (using google lol) the nearest prime number which is 67.